packages/cli/src/commands/setup/ui/libraries/windicss.js

```diff
@@ -0,0 +1,147 @@
+    type: 'boolean',
+  })
+
+  yargs.option('install', {
+    alias: 'i',
+    default: true,
+    description: 'Install packages',
+    type: 'boolean',
+  })
+}
+
+const windiImportsExist = (appTsx) => appTsx.match(/^import 'windi\.css'$/m)
+
+export const handler = async ({ force, install }) => {
+  const rwPaths = getPaths()
+
+  const packages = ['windicss-webpack-plugin', 'windicss']
+
+  const tasks = new Listr([
+    {
+      title: 'Installing packages...',
+      skip: () => !install,
+      task: () => {
+        return new Listr([
+          {
+            title: `Install ${packages.join(', ')}`,
+            task: async () => {
+              await execa('yarn', [
+                'workspace',
+                'web',
+                'add',
+                '-D',
+                ...packages,
+              ])
+            },
+          },
+        ])
+      },
+    },
+    {
+      title: 'Setup Webpack...',
+      task: () => {
+        return new Listr([
+          {
+            title: 'Setup Webpack',
+            task: async () => {
+              await execa('yarn', [
+                'redwood',
+                'setup',
+                'webpack'
+              ])
+            },
+          },
+          {
+            title: 'Configure WindiCSS',
+            task: async () => {
+              const webpackConfigPath = path.join(
+                rwPaths.web.config,
+                'webpack.config.js'
+              )
+              const webpackConfig = fs.readFileSync(webpackConfigPath, 'utf-8')
+              const newWebpackConfig = `const WindiCSSWebpackPlugin = require('windicss-webpack-plugin')\n`
+                + webpackConfig.replace(
+                  '// config.plugins.push(YOUR_PLUGIN)',
+                  '// config.plugins.push(YOUR_PLUGIN)\n  config.plugins.push(new WindiCSSWebpackPlugin())'
+                )
+              fs.writeFileSync(webpackConfigPath, newWebpackConfig)
+            },
+          },
+        ])
+      },
+    },
+    {
+      title: 'Initializing WindiCSS...',
+      task: async () => {
+        const windiConfigPath = path.join(
+          rwPaths.web.config,
+          'windi.config.js'
+        )
+
+        if (fs.existsSync(windiConfigPath)) {
+          if (force) {
+            fs.unlinkSync(windiConfigPath)
+          } else {
+            throw new Error(
+              'Windicss config already exists.\nUse --force to override existing config.'
+            )
+          }
+        }
+
+        const windiConfig = `
+          import { defineConfig } from 'windicss/helpers'
+
+          export default defineConfig({
+            extract: {
+              include: ['**/*.{jsx,tsx,css}'],
+              exclude: ['node_modules', '.git'],
+            },
+          })
+        `.split('\n').map((line) => line.trim()).join('\n')
+        fs.writeFileSync(windiConfigPath, windiConfig)
+      },
+    },
+    {
+      title: 'Adding import to App.tsx...',
```

Comment:
When I tested it locally, it failed at this step because I was on a JS project instead of a TS one. Instead of trying to find the file yourself, you can use `rwPaths.web.app`—that should always give you the file, no matter the extension.

Reply:
Oooh that's smart, I didn't know it was pointing to the ``App.tsx/App.jsx`` file! This is wonderful! I'm changing it right now!